src/forms/NumberInput.jsx

```diff
@@ -34,207 +34,29 @@ type Props = {
-			max,
-			min,
-			name,
-			onChange, // eslint-disable-line no-unused-vars
-			required,
-			requiredText,
-			step,
-			value, // eslint-disable-line no-unused-vars
-			...other
-		} = this.props;
-
-		const classNames = {
-			field: cx('field--reset span--100', { 'field--error': error }, className),
-			fauxInput: cx(FAUX_INPUT_CLASS, fauxInputClassName, {
-				disabled,
-				error,
-				[FOCUSED_INPUT_CLASS]: this.state.isFieldFocused,
-			}),
-			label: cx(
-				'label--field',
-				{
-					'label--disabled': disabled,
-					'label--required': required,
-					'flush--bottom': helperText,
-				},
-				labelClassName
-			),
-			helperText: cx('helperTextContainer', { required, disabled }),
-			incrementBtn: cx('button--reset', INCREMENT_BTN_CLASS),
-			decrementBtn: cx('button--reset', DECREMENT_BTN_CLASS),
-		};
+		const { id, helperText, label, required, requiredText, ...other } = this.props;
 
 		return (
 			<React.Fragment>
 				{label && (
-					<label
-						className={classNames.label}
-						htmlFor={id}
-						data-requiredtext={required && requiredText}
-					>
+					<FieldLabel htmlFor={id} data-requiredtext={required && requiredText}>
 						{label}
-					</label>
+					</FieldLabel>
 				)}
-				{helperText && <div className={classNames.helperText}>{helperText}</div>}
-				<div
-					className={classNames.fauxInput}
-					ref={(el: HTMLButtonElement | null) => (this.fauxInputEl = el)}
-				>
-					<Flex align="center">
-						<FlexItem>
-							<input
-								type="number"
-								pattern="[0-9]*"
-								id={id}
-								name={name}
-								max={max}
-								min={min}
-								step={step}
-								value={this.state.value === null ? '' : this.state.value}
-								required={required}
-								className={classNames.field}
-								onBlur={this.onBlur}
-								onFocus={this.onFocus}
-								onChange={this.onChange}
-								onKeyDown={this.onKeyDown}
-								disabled={disabled}
-								{...other}
-							/>
-						</FlexItem>
-
-						<FlexItem shrink>
-							<Button
-								reset
-								tabIndex="-1"
-								className={classNames.decrementBtn}
-								onBlur={this.onBlur}
-								onClick={this.decrementAction}
-								onFocus={this.onFocus}
-								ref={(el: HTMLButtonElement | null) =>
-									(this.decrementBtnEl = el)
-								}
-							>
-								<Icon shape="minus" size="xs" />
-							</Button>
-						</FlexItem>
-
-						<FlexItem shrink>
-							<Button
-								reset
-								tabIndex="-1"
-								className={classNames.incrementBtn}
-								onBlur={this.onBlur}
-								onClick={this.incrementAction}
-								onFocus={this.onFocus}
-								ref={(el: HTMLButtonElement | null) =>
-									(this.incrementBtnEl = el)
-								}
-							>
-								<Icon shape="plus" size="xs" />
-							</Button>
-						</FlexItem>
-
-						{children}
-					</Flex>
-				</div>
+				{helperText && <FieldHelper>{helperText}</FieldHelper>}
+				<SwarmNumeric
+					type="number"
+					pattern="[0-9]*"
```

Comment:
do we need to pass `type` and `pattern` to `SwarmNumeric`? That seems like something that should be handled internally

Reply:
Not needed, they're handled in `SwarmNumeric` as well.